pkg/internal/client/client.go

```diff
@@ -122,3 +131,66 @@ func (c *Client) VMISerialConsole(namespace, name string, timeout time.Duration)
 		&kubecli.SerialConsoleOptions{ConnectionTimeout: timeout},
 	)
 }
+
+func (c *Client) ExecuteCommandOnPod(ctx context.Context,
+	namespace, name, containerName string,
+	command []string) (stdout, stderr string, err error) {
+	resultCh := make(chan executeWrapper, 1)
+
+	go func() {
+		var (
+			stdoutBuf bytes.Buffer
+			stderrBuf bytes.Buffer
+		)
+		options := remotecommand.StreamOptions{
+			Stdout: &stdoutBuf,
+			Stderr: &stderrBuf,
+			Tty:    false,
+		}
+
+		err = executeCommandOnPodWithOptions(c.KubevirtClient, namespace, name, containerName, command, options)
+		stdout = stdoutBuf.String()
+		stderr = stderrBuf.String()
+		resultCh <- executeWrapper{stdout, stderr, err}
+	}()
+
+	select {
+	case result := <-resultCh:
+		return result.stdout, result.stderr, result.err
+	case <-ctx.Done():
+		return stdout, stderr, ctx.Err()
+	}
+}
+
+func executeCommandOnPodWithOptions(virtCli kubecli.KubevirtClient,
+	namespace, name, containerName string,
+	command []string,
+	options remotecommand.StreamOptions) error {
+	req := virtCli.CoreV1().RESTClient().Post().
+		Resource("pods").
+		Name(name).
+		Namespace(namespace).
+		SubResource("exec").
+		Param("container", containerName)
+
+	req.VersionedParams(&corev1.PodExecOptions{
+		Container: containerName,
+		Command:   command,
+		Stdin:     false,
+		Stdout:    true,
+		Stderr:    true,
+		TTY:       false,
+	}, scheme.ParameterCodec)
+
+	virtConfig, err := kubecli.GetKubevirtClientConfig()
+	if err != nil {
+		return err
+	}
```

Comment:
This configuration struct was built when we called `client.New()`, please consider saving it as a state.

Reply:
what do you mean?

Reply:
In `client.New()`, we call:
```go
	config, err := rest.InClusterConfig()
	if err != nil {
		return nil, err
	}
```

This `config` struct could be saved as a field in the Client struct, if it is needed again.

Reply:
DONE